src/app/AllowOrBlockDomainDialog.qml

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019 Chris Clime
+ *
+ * This file is part of morph-browser.
+ *
+ * morph-browser is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * morph-browser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.4
+import Ubuntu.Components 1.3
+import webbrowsercommon.private 0.1
+
+ModalDialog {
+    objectName: "allowOrBlockDomain"
+    title: i18n.tr("Block domain")
+
+    property string domain
+
+    message: i18n.tr("About to access domain %1. Do you want to block this domain from now on?".arg(domain));
```

Comment:
This text still seems unclear to me. As a user, I don't know what will happen if I choose each option. Since this dialog only appears when whitelist-only mode is enabled (I think), let's refine it as "You're trying to access *domain* but it is not on your approved domain whitelist. Would you like to continue to block the domain, allow browsing to the domain, or go back?"

Reply:
Maybe that's too long.

Reply:
yes it is right, the dialog does only appear in "whitelist-only" mode.

There are two types of requests though: mainFrame requests (navigating to a different page) and requests within a page to another domain.

For the non-mainFrame requests I would suggest sth. like: 
"_parentDomain_ is trying to access _domain_ but it is not on your approved domain whitelist. Would you like to continue to block the domain, allow loading data from the domain, or just ignore this request ?"

Reply:
How would I see the second type of request?

Reply:
e.g https://nasa.gov
those requests are async, so several dialogs will appear. But the order the user blocks them does not matter.

Reply:
Mmm... tested the latest changes. I think the experience could be polished a bit but I don't think is bad how it is now. How about going ahead how it is and revisit later on?

Reply:
by the way: the allow / block dialog was inspired from the "Jade Diamond" browser : )
![image](https://user-images.githubusercontent.com/848115/65726808-3d61cc80-e0b6-11e9-9848-f584803d1c01.png)

Reply:
Jade Diamond is not a core app  :D